subsys/bluetooth/host/id.c

```diff
@@ -286,6 +286,7 @@ static void le_rpa_invalidate(void)
 			return;
 		}
 		bool rpa_expired_data[bt_dev.id_count];
+		memset(rpa_expired_data, true, sizeof(rpa_expired_data));
```

Comment:
Mixing a `bool` array with `memset()` looks questionable to me, since it's architecture dependent what the in-memory representation of a `bool` is. Seems to me like this would be more compactly (and correctly) done with a bitmap instead of a bool array.

Reply:
That is correct, bool is architecture dependent, but the memset will just fill the array. I don't see any risk here.
But that is correct, the compactness is not ideal, but that would be out of scope of this bugfix.

Reply:
Alternatively we could switch the condition around and initialize it as ` = { 0 }`

Reply:
That is doable. @jhedberg would that be alright for you?

Reply:
Using an initializer is fine, but `{ 0 }` (which is the same as just `{}`) is not what you want since you want the elements to be true. You could use an array index range, but then you need to use a build-time constant for the array size instead of `bt_dev.id_count`, e.g.:

```c
	bool rpa_expired_data[CONFIG_BT_ID_COUNT] = {
		[0 ... (CONFIG_BT_ID_COUNT - 1)] = true,
	};
```

> the memset will just fill the array. I don't see any risk here

I was thinking of a situation where on some architecture the in-memory representation of `bool` is of a word size, e.g. 32 bits, and the value of `true` is `0x00000001`, however your memset would typecast that to `uint8_t` and set each byte to that value, so you'd get elements with `0x01010101` values.

Reply:
> Using an initializer is fine, but `{ 0 }` (which is the same as just `{}`) is not what you want since you want the elements to be true.

The suggestion was to reverse the checks, effectively making it `rpa_non_expired_data` or something. 

> You could use an array index range, but then you need to use a build-time constant for the array size instead of bt_dev.id_count

That may be a good solution nonetheless, to avoid using a variable value to stack-allocate an array (although I doubt that `bt_dev.id_count` will ever be that big).

Reply:
We could also just use a simple `for` loop to initialize the array to `true` rather than all these clever solutions :D

Reply:
Sorry, I missed the detail about reversing the condition. Then the simplest thing is to just do `= {};`